adserver/utils.py

```diff
@@ -249,22 +252,29 @@ def is_proxy_ip(ip):
     return False
 
 
-def get_geolocation(ip_address):
+def get_geolocation(request, ip_address):
+    """Gets the geolocation for this IP address."""
+    geolocation = {
+        # Use the Cloudflare GeoIP if available
+        "country_code": getattr(request, "user_country", None),
+        "region": None,
+        "dma_code": None,
+    }
     try:
         ipaddress.ip_address(force_text(ip_address))
     except ValueError:
         # Invalid IP address
-        return None
+        return geolocation
 
     if geoip:
         try:
-            return geoip.city(ip_address)
+            geolocation = geoip.city(ip_address)
         except AddressNotFoundError:
             log.debug("Could not get geolocation for %s", ip_address)
         except GeoIP2Exception:
             log.warning("Geolocation configuration error")
 
-    return None
+    return geolocation
```

Comment:
Ya, this doesn't quite seem right. Here's how I think it should work:

- If we get a country from Cloudflare, use that.
- Use the GeoIP database just to fill in the region/dma. Maybe remove this since we aren't using it at all?
- We probably don't need to validate the IP address at all here. We should assume that the request's IP is accurate from a middleware.

Reply:
I updated this to just always use what the middleware sets.